src/riscv/lib/src/pvm/linux/fs.rs

```diff
@@ -23,6 +23,32 @@ impl<M: ManagerBase> SupervisorState<M> {
         Err(Error::Access)
     }
 
+    /// Handle `close` system call. All access to the file system is denied.
+    ///
+    /// See <https://man7.org/linux/man-pages/man2/close.2.html>
+    pub(super) fn handle_close(&mut self) -> Result<u64, Error>
+    where
+        M: ManagerReadWrite,
+    {
+        // None of the file descriptors we allow (stdout/stderr) can sensibly be closed.
+        Err(Error::Access)
+    }
+
+    /// Handle `read` system call. All access to the file system is denied.
+    ///
+    /// See <https://man7.org/linux/man-pages/man2/read.2.html>
+    pub(super) fn handle_read(&mut self, length: u64) -> Result<u64, Error>
+    where
+        M: ManagerReadWrite,
+    {
+        if length == 0 {
+            // If the length is zero then the standard allows returning zero without reading.
+            return Ok(0);
+        }
```

Comment:
This may be surprising to some, despite the reference to the man page. Could you add a small note explaining that POSIX allows omission of error checking if the length is zero?